qa/evil-tests/src/test/java/org/elasticsearch/plugins/InstallPluginCommandTests.java

```diff
@@ -477,6 +477,37 @@ public void testBinPermissions() throws Exception {
         }
     }
 
+    public void testPlatformBinPermissions() throws Exception {
+        assumeTrue("posix filesystem", isPosix);
+        Tuple<Path, Environment> env = createEnv(fs, temp);
+        Path pluginDir = createPluginDir(temp);
+        Path platformDir = pluginDir.resolve("platform");
+        Path platformNameDir = platformDir.resolve("linux-x86_64");
+        Path platformBinDir = platformNameDir.resolve("bin");
+        Files.createDirectories(platformBinDir);
+        Path programFile = Files.createFile(platformBinDir.resolve("someprogram"));
+        try (PosixPermissionsResetter binAttrs = new PosixPermissionsResetter(programFile)) {
+            Set<PosixFilePermission> perms = binAttrs.getCopyPermissions();
+            // make sure execute permissions are missing (they should be by default but just in case)
+            perms.remove(PosixFilePermission.OWNER_EXECUTE);
+            perms.remove(PosixFilePermission.GROUP_EXECUTE);
+            perms.remove(PosixFilePermission.OTHERS_EXECUTE);
+            binAttrs.setPermissions(perms);
```

Comment:
I changed the test slightly so it asserts the permissions both before and after install